### ISSUE TITLE
cg_state_t refactoring

### ISSUE THREAD

**Issue by jdolan** [jdolan/quetoo#532]
Taken from comments on a closed PR:

cg_state_t is a horrible name for what are really bg_pmove parameters. That struct should be renamed to something more appropriate.

Likewise, the getters for it are unnecessarily wordy. Cg_GetHookPullSpeed and Cg_GetCrouchSlideBoost etc.. I'm not even sure why these exist. There should just be an extern cg_pm_params_t or something that contains these things as members directly.

Similarly, I think that, rather than chew through config strings for each of these attributes, we should munge them all into a single config string that is user-info-delimited (\\):

    gravity
    hook pull speed
    crouch slide enabled
    crouch side boost (a value of -1 could also serve to disable couch sliding?)
    crouch slide friction
    etc..

These variables will change so infrequently, that re-transmitting all of them in one string will be fine. That way, we're not burning config strings on things that almost never change. Using the user-info format will allow us to easily marshal and parse these values with existing code.

**Comment by kaadmy** [comment]
I think it should not be named specifically for pmove since other game configuration could be here (unless there's a better place for them) such as weapon refire times for client-side weapon ready feedback or similar.

**Comment by jdolan** [comment]
cg_config_t perhaps, then? The point was more to move away from getter methods for everything, and instead just lump these things together into a single ConfigString. When that ConfigString changes (rarely) re-parse it, and extract all of these members directly into a struct. That's really the point of the ticket. Worth clarifying, tho. Thanks!

**Comment by jdolan** [comment]
cg_game_t might also be appropriate. A lot of these variables will come from g_game_t or g_level_t on the server.

**Comment by kaadmy** [comment]
`cg_game_config_t`?

**Comment by jdolan** [comment]
It looks like the majority of these things will correspond to fields in `g_level_t` in the game module; the idea being that these things are expected to change _usually_ only when the level itself changes or is restarted. So `cg_level_t` probably makes the most sense.

**Comment by jdolan** [comment]
If we wanted to extract members from `g_level_t` into a new `g_config_t`, and call this new cgame struct `cg_config_t`, I'm also totally fine with that. You can also just leave this one on my plate if you want to :)

**Comment by jdolan** [comment]
Let's hold off on addressing this until after #487 is closed (or address it in that branch).

**Comment by jdolan** [comment]
After talking this out on Discord, it probably makes the most sense to break _all_ pmove scalars out into a struct that is passed into Pm_Move on both the client and the server. This struct should be dictated by the server, and sent to all clients whenever any of its members change. This would also allow us to then cvar-ify everything from air acceleration to crouch speed in the game module. Doing so would allow folks to create "custom" movement profiles to e.g. recreate Quake1 physics -- very cool for some of our classic Quake1 map remakes.

**Comment by jdolan** [comment]
Closing this out and will reopen another ticket for pmove params struct.
